cmd/snap/cmd_run_test.go

```diff
@@ -1867,377 +2420,68 @@ func openHintFileLock(snapName string) (*osutil.FileLock, error) {
-			TimeRemaining: 0,
-		})
-		return nil
-	})
-	defer restorePendingRefreshNotification()
-
-	var finishRefreshNotificationCalled int
-	restoreFinishRefreshNotification := snaprun.MockFinishRefreshNotification(func(ctx context.Context, refreshInfo *usersessionclient.FinishedSnapRefreshInfo) error {
-		finishRefreshNotificationCalled++
-		c.Check(refreshInfo, check.DeepEquals, &usersessionclient.FinishedSnapRefreshInfo{
-			InstanceName: "some-snap",
-		})
-		return fmt.Errorf("boom")
-	})
-	defer restoreFinishRefreshNotification()
-
-	inhibitInfo := runinhibit.InhibitInfo{Previous: snap.R(11)}
-	c.Assert(runinhibit.LockWithHint("some-snap", runinhibit.HintInhibitedForRefresh, inhibitInfo), check.IsNil)
-
-	restore := snaprun.MockWaitWhileInhibited(func(ctx context.Context, snapName string, notInhibited func(ctx context.Context) error, inhibited func(ctx context.Context, hint runinhibit.Hint, inhibitInfo *runinhibit.InhibitInfo) (cont bool, err error), interval time.Duration) (flock *osutil.FileLock, retErr error) {
-		c.Check(snapName, check.Equals, "some-snap")
-
-		cont, err := inhibited(ctx, runinhibit.HintInhibitedForRefresh, nil)
-		c.Assert(err, check.IsNil)
-		// non-service apps should keep waiting
-		c.Check(cont, check.Equals, false)
-		if notInhibited != nil {
-			c.Errorf("this should never be reached")
-		}
-
-		flock, err = openHintFileLock(snapName)
-		c.Assert(err, check.IsNil)
-		return flock, nil
-	})
-	defer restore()
-
-	c.Assert(snaprun.WaitWhileInhibited(context.TODO(), "some-snap"), check.ErrorMatches, "boom")
-
-	c.Check(pendingRefreshNotificationCalled, check.Equals, 1)
-	c.Check(finishRefreshNotificationCalled, check.Equals, 1)
-
-	// lock must be released
-	checkHintFileNotLocked(c, "some-snap")
-}
-
-func (s *RunSuite) TestWaitWhileInhibitedContextCancellationOnError(c *check.C) {
-	inhibitInfo := runinhibit.InhibitInfo{Previous: snap.R(11)}
-	c.Assert(runinhibit.LockWithHint("some-snap", runinhibit.HintInhibitedForRefresh, inhibitInfo), check.IsNil)
-
-	restoreIsGraphicalSession := snaprun.MockIsGraphicalSession(true)
-	defer restoreIsGraphicalSession()
-
-	originalCtx, cancel := context.WithCancel(context.Background())
-	restoreTryNotifyRefresh := snaprun.MockTryNotifyRefreshViaSnapDesktopIntegrationFlow(func(ctx context.Context, snapName string) bool {
-		c.Check(snapName, check.Equals, "some-snap")
-		// check context is propagated properly
-		c.Assert(ctx, check.Equals, originalCtx)
-		c.Check(ctx.Err(), check.IsNil)
-		// cancel context to trigger cancellation error
-		cancel()
-		return true
-	})
-	defer restoreTryNotifyRefresh()
-
-	err := snaprun.WaitWhileInhibited(originalCtx, "some-snap")
-	c.Assert(err, check.ErrorMatches, "context canceled")
-	c.Assert(errors.Is(err, context.Canceled), check.Equals, true)
-	c.Assert(errors.Is(originalCtx.Err(), context.Canceled), check.Equals, true)
-}
+// func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlowErrorOnFinish(c *check.C) {
```

Comment:
Looks like lots of commented-out code below here. Is this on purpose?

Reply:
Thank for catching this, I forgot to remove it.